etl/generative_model_pipelines/generative_model_pipeline.py

```diff
@@ -0,0 +1,14 @@
+from abc import ABC, abstractmethod
+
+from etl.models import Record
+
+
+class GenerativeModelPipeline(ABC):
```

Comment:
Nit: I might call this `RecordEnrichmentPipeline` or similar. The interface name should suggest what task the interface is doing, not how it's doing it.